### PR TITLE
make `app` mandatory in a GuStack

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

For now, we do not support a multi-app stack so make `app` mandatory for simplicity.

## Does this change require changes to existing projects or CDK CLI?

Yes, as the signature of GuStack's constructor has changed.

## How to test

n/a

## How can we measure success?

Simpler code, for now?

## Have we considered potential risks?

We'll need to support multi-app stacks at some point, so this might cause a massive diff later.

## Images

n/a

### CODE REVIEW

File: test/utils/simple-gu-stack.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { App } from "@aws-cdk/core";
+import { GuStack } from "../../src/constructs/core";
+
+export const simpleGuStackForTesting = () => new GuStack(new App(), "Test", { app: "testing" });
```
Comment (jamie-lynch) [inline test/utils/simple-gu-stack.ts:4]: This is a great idea! Maybe it should be it's own PR though as it's a pretty big diff?

Comment (akash1810) [reply test/utils/simple-gu-stack.ts]: Good idea - https://github.com/guardian/cdk/pull/70

**Review by jamie-lynch — APPROVED** [review]
